build/BuildScript.cs

```diff
@@ -16,44 +16,30 @@ public class BuildScript : DefaultBuildScript
         [SolutionFileName]
         public string SolutionFileName { get; set; } = "HttpReports.sln";
 
-        [FromArg("c")]
+        [FromArg("c", "Build configuration")]
         [BuildConfiguration]
         public string Configuration { get; set; } = "Release";
 
-        public string Version { get; set; } 
+        [FromArg("version", "Build version")]
+        public string Version { get; set; }
 
-        public string NugetKey = "";
+        [FromArg("key", "nuget key for publishing nuget packages.")]
+        public string NugetKey { get; set; } = string.Empty;
 
         public FullPath OutputDir => RootDirectory.CombineWith("output");
 
         public FullPath SourceDir => RootDirectory.CombineWith("src");
 
         protected override void ConfigureTargets(ITaskContext context)
-        { 
-            if (context.ScriptArgs.ContainsKey("version"))
-            {
-                Version = context.ScriptArgs["version"];
-            }
-
-            if (context.ScriptArgs.ContainsKey("key"))
-            {
-                NugetKey = context.ScriptArgs["key"];
-            }  
-
+        {
             context.LogInfo("============================================"); 
             context.LogInfo($"NugetKey:{NugetKey} Version:{Version}");
             context.LogInfo("============================================"); 
 
             var clean = context.CreateTarget("Clean")
-                .SetDescription("Clean's the solution") 
-                
-                .AddCoreTask(x => { 
-                    
-                        context.GetFiles(OutputDir, "*.*").Where(c => !c.FileName.Contains(Version)).ToList().ForEach(t => System.IO.File.Delete(t)); 
```

Comment:
You should not add custom code to AddTask or AddCoreTask as it will get executed everytime a target is executd by flubu. Instead i've added AddDirectoryToClean. We could also move code to Do method similar as I did it with nuget push.